### PR TITLE
Add documentation for ADS1118

### PR DESCRIPTION
**Pull request by solomondg1**
## Description:

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#5711

## Checklist:

  - [1 ] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [1 ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Review by lboue — COMMENTED** [review]
Looks good

File: components/sensor/ads1118.rst [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+.. _ads1118-sensor:
+
+Sensor
+------
+
+The ``ads1118`` sensor allows you to use your ADS1118 delta-sigma ADC
+sensors (`datasheet <https://www.ti.com/lit/ds/symlink/ads1118.pdf>`__) with ESPHome.
+First, setup an :ref:`ADS1118 Hub <ads1118-component>` for your ADS1118 sensor and then use this
+sensor platform to create individual sensors that will report the voltage to Home Assistant.
```
Comment (jesserockz) [inline components/sensor/ads1118.rst]: While we love Home Assistant, it's not needed on individual sensor pages. =)
```suggestion
sensor platform to create individual sensors that will report the voltage.
```

File: components/sensor/ads1118.rst [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+************************
+
+-  **type** (*Optional*): ``adc``(default) or ``temperature``
+-  **name** (**Required**, string): The name for this sensor.
+-  **ads1118_id** (*Optional*, :ref:`config-id`): Manually specify the ID of the
+   :ref:`ADS1118 Hub <ads1118-component>` you want to use this sensor.
+-  **update_interval** (*Optional*, :ref:`config-time`): The interval to check the sensor. Defaults to ``60s``.
+-  **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
+
+adc options:
```
Comment (jesserockz) [inline components/sensor/ads1118.rst]:
```suggestion
ADC Options:
^^^^^^^^^^^^
```

File: components/sensor/ads1118.rst [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+-  **multiplexer** (**Required**): The multiplexer channel of this sensor. Effectively means between which pins you want to measure voltage.
+-  **gain** (**Required**, float): The gain of this sensor.
+
+temperature options: none
```
Comment (jesserockz) [inline components/sensor/ads1118.rst]:
```suggestion
Temperature Options:
^^^^^^^^^^^^^^^^^^^^

None
```

File: components/sensor/ads1118.rst [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        id: 'batt_volt' 
+        name: "Battery Voltage"
+        multiplexer: 'A3_GND'
+        gain: 4.096
+        update_interval: .01s
+        filters:
+        - sliding_window_moving_average:
+            window_size: 1000
+            send_every: 1000
+        - lambda: return x * 28.3 / 6.5;
```
Comment (jesserockz) [inline components/sensor/ads1118.rst]: We don't add things like this to sensor docs. It is up to the user to find out what they need exactly from the filter documentation
```suggestion
```

File: components/sensor/ads1118.rst [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+Configuration variables:
+************************
+
+-  **type** (*Optional*): ``adc``(default) or ``temperature``
+-  **name** (**Required**, string): The name for this sensor.
+-  **ads1118_id** (*Optional*, :ref:`config-id`): Manually specify the ID of the
+   :ref:`ADS1118 Hub <ads1118-component>` you want to use this sensor.
+-  **update_interval** (*Optional*, :ref:`config-time`): The interval to check the sensor. Defaults to ``60s``.
+-  **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
```
Comment (jesserockz) [inline components/sensor/ads1118.rst]:
```suggestion
- **type** (*Optional*): ``adc``(default) or ``temperature``
- **ads1118_id** (*Optional*, :ref:`config-id`): Manually specify the ID of the
   :ref:`ADS1118 Hub <ads1118-component>` you want to use this sensor.
- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the sensor. Defaults to ``60s``.
- All other options from :ref:`Sensor <config-sensor>`.
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Review by jesserockz — APPROVED** [review]

**Pull request merged**